project/frontend/src/components/Authentication.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react';
+
+const AuthContext = React.createContext({});
+
+function Authentication(props) {
+  const [GoogleAuth, setGoogleAuth] = useState(undefined);
+  const [currentUser, setCurrentUser] = useState(undefined);
+  const [signedIn, setSignedIn] = useState(false);
+
+  const context = {
+    GoogleAuth: { get: GoogleAuth, set: setGoogleAuth },
+    currentUser: { get: currentUser, set: setCurrentUser },
+    signedIn: { get: signedIn, set: setSignedIn },
+  };
+
+  useEffect(() => {
+    const onGoogleAuthLoaded = () => {
+      window.gapi.load('auth2', () => {
```

Comment:
Add to a try, catch block

Reply:
To be honest I don't think it is possible that gapi is not defined by the time this effect is running. I am certain now that defer scripts are run before DOMContentLoaded and according to some testing, I don't think this effect will ever run before DOMContentLoaded.

If you still think I should use a try/catch though, let me know and I will add it.